test/util/test_path_provider.py

```diff
@@ -0,0 +1,173 @@
+import os
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+
+class TestPathProviderBase(ABC):
+    """
+        used to generate the files path's in all tests
+        in particular this file is helping with the integration between running test_integration script
+        and the integration_test_runner.
+        all test path's should go through here.
+        in the future should use the constants module to be more stable.
+        we use abstract class so that every enviorment could easily setup it's own files structure.
+    """
+
+    def __init__(self, root_dir, file_helper_dir):
+        self.ROOT_DIR = root_dir
+        self.HELPERS_FILES_DIR = file_helper_dir
+
+    @property
+    def conjurrc_path(self):
+        return os.path.join(self.ROOT_DIR, self.conjurrc_file_name)
+
+    @property
+    def netrc_path(self):
+        return os.path.join(self.ROOT_DIR, self.netrc_file_name)
+
+    @property
+    def pem_file_path(self):
+        return os.path.join(self.ROOT_DIR, self.pem_file_name)
+
+    @property
+    def conjurrc_file_name(self):
+        return '.conjurrc'
+
+    @property
+    @abstractmethod
+    def test_conjurrc_file_path(self):
+        raise NotImplemented()
+
+    @property
+    def netrc_file_name(self):
+        return '.netrc'
+
+    @property
+    def pem_file_name(self):
+        return 'conjur-server.pem'
+
+    @property
+    def helpers_files_path(self):
+        return self.HELPERS_FILES_DIR
+
+    @property
+    @abstractmethod
+    def nginx_conf_path(self):
+        raise NotImplemented()
+
+    @abstractmethod
+    def get_policy_path(self, type: str):
+        raise NotImplemented()
+
+
+class TestRunnerPathProvider(TestPathProviderBase):
+    __instance = None
+
+    @staticmethod
+    def getInstance():
+        """
+            Implement the singelton design pattern. so we will be sure all path's are stable
```

Comment:
"So" and "paths"

What do you mean by stable here?

Reply:
During development I notice few componnents could call the TestProvider. If they call it with different configs than the same code could be called in different areas but produce different values.
The singelton pattern means that only it could be only initialized in one place and all consumers will use the same configurations.
Now that we talk I think that after we changed to using TestPathProviderBase. I think it will be better that all will use it getinstance method that will return the desired implementation